### ISSUE TITLE
Can't get multiple repos to work

### ISSUE THREAD

**Issue by yostie** [ytti/oxidized#387]
I am trying to segment out configs into general network infrastructure and security configs but I can't seem to get the oxidized config file correct to do multiple repos.  Are there formal docs for Oxidized that I am missing somewhere besides in the Github info?  Here is the error I get followed by the code:

I, [2016-04-08T16:40:14.921835 #11972]  INFO -- : Oxidized starting, running as pid 11972
resolving DNS for 10.4.252.62...
F, [2016-04-08T16:40:15.001342 #11972] FATAL -- : Oxidized crashed, crashfile written in /etc/oxidized/crash
can't convert Asetus::ConfigStruct to String (Asetus::ConfigStruct#to_str gives Asetus::ConfigStruct)
/home/mikey/.rvm/gems/ruby-2.1.2/gems/oxidized-0.13.1/lib/oxidized/output/git.rb:24:in `expand_path': can't convert Asetus::ConfigStruct to String (Asetus::ConfigStruct#to_str gives Asetus::ConfigStruct) (TypeError)

```yaml
---
username: omitted
password: omitted
model: ios
interval: 3600
use_syslog: false
debug: true
threads: 30
timeout: 20
retries: 3
prompt: !ruby/regexp /^([\w.@-]+[#>]\s?)$/
rest: omitted:8888
vars: {}
groups: { }        

input:
  default: ssh
  debug: false
  ssh:
    secure: false

output:
  default: git
  git:
    user: omitted
    email: omitted
    repo:
      network: "~/configs/network.git"
      security: "~/configs/security.git"

source:
  default: csv
  csv:
    file: ~/.config/oxidized/router.db
    delimiter: !ruby/regexp /:/
    map:
      name: 0
      model: 1


model_map:
  cisco: ios
  juniper: junos
```

**Comment by ytti** [comment]
@danilopopeye can probably elaborate on the

```
repo:
  one: x
  two: z
```

I think those are specific to the github hook. The output git wants repo as string. So you'd havejust

```
output:
  git:
    repo: ~/configs/main.git
```

Then you'd have groups for nodes, if node does not have any group, it'll go to 'main.git' repo, if it does have group it'll go to 'group.git' in that directory.

**Comment by yostie** [comment]
Sorry, I think I'm still missing something here.  Whenever I try to write the repos like it shows in the readme, which matches your first example above except specifies the directory, it crashes.  If I write it like you have it in your second example, it works fine, but then i don't have multiple repos.  In the groups section, I tried doing this to see if it helps, but still doesn't work.  Can you show a full example that would have multiple repos?  I think I'm missing something in the routers.db to tie the device to a group too.  Is that as simple as writing the routers.db and config like this?

routers.db
--------------
10.10.10.10:nxos:network
20.20.20.20:asa:security

config
--------
```
groups: { network, security }

... omitted for brevity

output:
  default: git
  git:
    user: omitted
    email: omitted
    repo:
       network: my/path/network.git
       security: my/path/security.git
```

**Comment by yostie** [comment]
I figured it out...  I was missing the modification in the routers.db.  For future dummies like me, this is what ended up working for me.  In routers.db I added another field for the group and then in the source section I added an entry for group: 3.  Last, I went with the single repo option.  What is the "groups" entry in the top section of the YAML file used for?  Looking through all the other issues people have had, nobody has used that groups dictionary entry.  

This ended up creating a network directory and a security directory within the same repo which is perfect.

routers.db
```
10.10.10.10:nxos:network
20.20.20.20:nxos:security
```

config
```
---
username: omitted
password: omitted
model: ios
interval: 3600
use_syslog: false
debug: true
threads: 30
timeout: 20
retries: 3
prompt: !ruby/regexp /^([\w.@-]+[#>]\s?)$/
rest: omitted:8888
vars: {}
groups: {} 

input:
  default: ssh
  debug: false
  ssh:
    secure: false

output:
  default: git
  git:
    user: omitted
    email: omitted
    single_repo: true
    repo: /my/path/network_configs.git

source:
  default: csv
  csv:
    file: ~/.config/oxidized/router.db
    delimiter: !ruby/regexp /:/
    map:
      name: 0
      model: 1
      group: 2

model_map:
  cisco: ios
  asa: asa
```

**Comment by yostie** [comment]
I have another question that hopefully someone can help with.  I started refining my desired directory structure and decided I want something like:

/my/path/internal/network
/my/path/internal/security
/my/path/cust_prem/customer

If I have the following relevant code:

config

```
output:
  single_repo: true
  repo: /my/path/network_configs.git

source:
  csv:
    map:
      name: 0
      model: 1
      group: 2
```

routers.db

```
10.10.10.10:nxos:internal/network
20.20.20.20:asa:cust_prem/security
```

oxidized runs and and does more than I want.  It will create those repos but it will create duplicate  configs.  It will create:

/my/path/internal/10.10.10.10
/my/path/internal/network/10.10.10.10
/my/path/cust_prem/20.20.20.20
/my/path/cust_prem/security/20.20.20.20

What am I doing wrong?

**Comment by ytti** [comment]
To be honest I have no idea what is going on there. But having multi-hierarchy groups isn't something that I've considered or tested. I do see how it could be useful though.

**Comment by yostie** [comment]
Yea my goal would be to separate my companies configs from customer configs.  Within my company's directory, i would break the configs out into device type (network or security) or possibly location.  For customers, i would likely have a folder for each customer.

New feature maybe?

**Comment by ytti** [comment]
Yup, I think this would be useful to support.

**Comment by danilopopeye** [comment]
> @danilopopeye can probably elaborate on the

Oh sorry, my bad. I was implementing this but didn't finished and forgot the docs. :astonished: 

I'm sending a PR fixing the docs in a minute. :+1:

**Comment by ytti** [comment]
This is fixed by 0.14.0 thanks all!
